AmplifyPlugins/Predictions/AWSPredictionsPlugin/Support/Utils/IdentifyTextResultTransformers.swift

```diff
@@ -52,6 +52,7 @@ class IdentifyTextResultTransformers: IdentifyResultTransformers {
                                   identifiedLines: identifiedLines)
     }
 
+    // swiftlint:disable cyclomatic_complexity
```

Comment:
Instead of disabling swiftlint, I think we should be using it as a hint to simplify our code.  I looked into this, and seems that adding the guard statements is forcing our cyclomatic complexity for this function to go up.  I was able to simplify the code so that we do not need this warning (if you can find out a better/different way to do it, that's fine too.)

I made the following changes:
1.  To avoid `guard let identifier = block.identifier else {` in this function, we just filter out all textractTextBlocks that have an invalid identifier.
2.  To avoid multiple guard/let/else, create sub functions to pass back fully constructed objects, or nil in the event that the block doesn't have enough information
3.  Create additional function to processTables
4.  Create additional function to processKeyValues.

As a result, I ended up with this for `processText(_ textractTextBlocks: [AWSTextractBlock])`:
```
    static func processText(_ textractTextBlocks: [AWSTextractBlock]) -> IdentifyDocumentTextResult {

        var words = [IdentifiedWord]()
        var lines = [String]()
        var linesDetailed = [IdentifiedLine]()
        var selections = [Selection]()
        var fullText = ""
        var blockMap = [String: AWSTextractBlock]()
        var tableBlocks = [AWSTextractBlock]()
        var keyValueBlocks = [AWSTextractBlock]()

        let validTextractTextBlocks = textractTextBlocks.filter { block in
            !(block.identifier ?? "").isEmpty
        }

        for block in validTextractTextBlocks {
            let identifier = block.identifier!
            switch block.blockType {
            case .line:
                if let line = parseLineBlock(block: block) {
                    lines.append(line.text)
                    linesDetailed.append(line)
                }
            case .word:
                if let word = parseWordBlock(block: block) {
                    fullText += word.text + " "
                    words.append(word)
                    blockMap[identifier] = block
                }
            case .selectionElement:
                if let selection = parseSelectionElementBlock(block: block) {
                    selections.append(selection)
                    blockMap[identifier] = block
                }
            case .table:
                tableBlocks.append(block)
            case .keyValueSet:
                keyValueBlocks.append(block)
                blockMap[identifier] = block
            default:
                blockMap[identifier] = block
            }
        }

        let tables = processTables(tableBlocks: tableBlocks, blockMap: blockMap)
        let keyValues = processKeyValues(keyValueBlocks: keyValueBlocks, blockMap: blockMap)

        return IdentifyDocumentTextResult(
            fullText: fullText,
            words: words,
            rawLineText: lines,
            identifiedLines: linesDetailed,
            selections: selections,
            tables: tables,
            keyValues: keyValues)
    }
```

Constructing functions:
```
    static func parseLineBlock(block: AWSTextractBlock) -> IdentifiedLine? {
        guard let text = block.text,
            let boundingBox = processBoundingBox(block.geometry?.boundingBox),
            let polygon = processPolygon(block.geometry?.polygon) else {
                return nil
        }

        return IdentifiedLine(text: text,
                              boundingBox: boundingBox,
                              polygon: polygon,
                              page: Int(truncating: block.page ?? 0))
    }

    static func parseWordBlock(block: AWSTextractBlock) -> IdentifiedWord? {
        guard let text = block.text,
            let boundingBox = processBoundingBox(block.geometry?.boundingBox),
            let polygon = processPolygon(block.geometry?.polygon) else {
                return nil
        }

         return IdentifiedWord(text: text,
                               boundingBox: boundingBox,
                               polygon: polygon,
                               page: Int(truncating: block.page ?? 0))
    }

    static func parseSelectionElementBlock(block: AWSTextractBlock) -> Selection? {
        guard let boundingBox = processBoundingBox(block.geometry?.boundingBox),
            let polygon = processPolygon(block.geometry?.polygon) else {
                return nil
        }
        let selectionStatus = block.selectionStatus == .selected ? true : false
        return Selection(boundingBox: boundingBox, polygon: polygon, isSelected: selectionStatus)
    }
```

Process functions which return `[Table]` and `[AWSTextractBlock]`:
```
    static func processTables(tableBlocks: [AWSTextractBlock],
                              blockMap: [String: AWSTextractBlock]) -> [Table] {
        var tables = [Table]()
        if !tableBlocks.isEmpty {
            for tableBlock in tableBlocks {
                if let table = processTable(tableBlock, blockMap: blockMap) {
                    tables.append(table)
                }
            }
        }
        return tables
    }

    static func processKeyValues(keyValueBlocks: [AWSTextractBlock],
                                 blockMap: [String: AWSTextractBlock]) -> [BoundedKeyValue] {
        var keyValues =  [BoundedKeyValue]()
        if !keyValueBlocks.isEmpty {
            for keyValueBlock in keyValueBlocks where keyValueBlock.entityTypes?.contains("KEY") ?? false {
                if let keyValue = processKeyValue(keyValueBlock, blockMap: blockMap) {
                    keyValues.append(keyValue)
                }
            }
        }
        return keyValues
    }
```

Not ideal, but I think it at least breaks things up a bit more so that `processText` isn't so hairy.  Thoughts?

Reply:
This is a good idea and I will take it. 
But I should actually make this one back to a Draft PR, next week I need more time working on this PR, I believe the code is still missing some behaviors. I am creating an Android App to see how that looks like and also need to sync with Android.